src/librustc/ty/sty.rs

```diff
@@ -1780,6 +1780,13 @@ impl<'a, 'gcx, 'tcx> TyS<'tcx> {
         }
     }
 
+    pub fn is_pointer_sized(&self) -> bool {
```

Comment:
This name doesn't feel ideal. I'd rather have `is_target_agnostic_primitive` that return `true` only for non-{u,i}size integer types, `char` and floating-point types.

Reply:
Yeah, I was a bit undecided myself. That works better.

Reply:
It seems like reversing the sense of this function would complicate some of the tests below, in particular the one that checks `(is_pointer_sized() && !feature.pointer_sized)` or whatever. I think the function is ok "as is" though. I mean the `isize` and `usize` pattern are meant (I think) to evoke the "native size" of a pointer. I might prefer `is_pointer_sized_int` though, just to be a bit more specific (`is_target_dependent_int` would be ok too but feels less obvious to me).

Reply:
Yes, the function is used explicitly for the pointer-sized integers, so making it more general would obfuscate the uses. I can make the name a bit more descriptive as suggested, though.